content/post/the-blacker-the-box.md

```diff
@@ -0,0 +1,74 @@
+The machine learning community spends the bulk of its time working on and talking about fast feedback problems. Problems with fast feedback are defined by 1) the ability to quickly evaluate the correctness of a prediction[^1] and 2) the ability to play the game near infinite amounts of time[^2]. Some examples of fast feedback problems are:
+
+* Chess: it's easy to verify which player has one or lost. Feedback takes only as long as the length of the game.
+* Conversion for an Ad Placement: Feedback to Google or Facebook on whether you clicked a given advertisement, and whether you subsequently converted[^3] is nearly instantaneous.
+* Movie Recommendations: For a given list of potential movies to watch, Netflix gets near instantaneous feedback when you do or do not watch some of the content they've elevated for you.
+
+[Tech-debt considerations](https://ai.google/research/pubs/pub43146) aside, for fast-feedback problems the box can be impenetrably black *as long as you have confidence in your ability to accurately measure the prediction precision.* 
+
+If I am designing an application for optimizing landing-page content for my e-commerce site (i.e., choose the content that converts best), then I do not care if my data scientist has rigged up a prediction pipeline that involves passing a SVM prediction through [an octopus](https://en.wikipedia.org/wiki/Paul_the_Octopus) so long as that model out-performs every other model we've tested in our production environment.
+
+The key to this scenario is that I'm able to quickly and easily evaluate the performance of candidate prediction models and compare the current production model to new candidate models for evaluation. Because I have an *objective* measure of predictive success, I do not need any understanding of what the model is doing under-the-hood in order to make use of it.
+
+## Slow Feedback
+
+Unfortunately, not all of the problems we face as an Analytics team have fast-feedback properties. Very often, we need to make predictions about what will happen over incredibly long time horizons or we need to make one-off decisions that could have large impacts on the future health of our business. Some examples of slow feedback problems are:
+
+* Estimating customer lifetime value
+* Predicting customer re-order dates
+* Deciding where to open new retail locations
+* Attributing marketing spend to actual sales
+* Predicting the outcomes of elections[^4]
+
+For these types of problems, a purely black-box prediction model is rarely the best approach. The reason why is that you, as the analyst, probably have more information about the underlying phenomenon than are encoded in the data that model would have access to. 
+
+If you try to apply a deep-net to your LTV data, you'll probably get lots of very precise predictions but your chance of over-fitting the model will be very high, and you'll have a very limited ability to predict customer LTV beyond the time periods that are represented in your data set.
```

Comment:
Isn't the over fitting problem a function of (1) the amount of data you have and (2) the complexity of the model?

Reply:
I was thinking more in terms of "lack of generalizability" which does have to do with the amount of data you have, but maybe not as directly as I tend to think when I think about "amount of data".

Your LTV model might not generalize well if, for example, you release a new product that your customers can use. There's no amount of historical data that you could collect which would ever help you incorporate that into your model, so I don't think of it as an "amount of data" problem so much as a generalization-failure issue.

Does that make sense? Thoughts on helping to clarify this much appreciated (@ilanman )

Reply:
Maybe I should just say 

> you will probably get lots of very accurate in-sample predictions, but your model will likely not generalize very well to the future if your business model or competitive environment shifts at all.

?

Reply:
This paragraph is absolutely true (and echos the section `Training data != Future data` of my second post  :) )  

Some thoughts that may or may not be helpful:

I suppose the question then is: What is the benefit of having a model that you can interrogate vs. a black box model when you're faced with, effectively, no data from which to generalize. I suppose a black-box model is almost completely useless whereas an LTV model like you suggest can have as a starting point:

1. Prior domain knowledge as an input (what do experts believe will be the impact of the new product?)
2. A model that is apriori "reasonable" - e.g., retention, frequency and monetary value. Even if it hasn't been validated by new data yet.
3. Update as necessary with new, useful data.

A black box model probably makes it difficult to learn from small (relevant) data since you can't really tear it apart and know why it does what it does.  The above scheme is very "Bayesian" (and in fact Fader's models are basically an application of Empirical Bayes).